spec/controllers/products_controller_spec.rb

```diff
@@ -0,0 +1,314 @@
+require 'rails_helper'
+
+RSpec.describe ProductsController, type: :controller do
+  before do
+    @creator = create(:user)
+    @category = create(:category)
+    create_list(:product, 2, user_id: @creator.id, category_id: @category.id)
+    create_list(:product, 2, status: 'archived', user_id: @creator.id, category_id: @category.id)
+    create_list(:product, 2, status: 'published', user_id: @creator.id, category_id: @category.id)
+  end
+  describe 'GET #index' do
+    before do
+      @user = create(:user)
+      @product = create(:product, user_id: @user.id, status: 'published')
+      create(:product, user_id: @user.id)
+    end
+
+    context 'when a user is signed in' do
+      before do
+        sign_in(@user)
+        get :index
```

Comment:
You can add this line to the before block

Reply:
The user needs to sign in before calling the index or the test won't pass!

Reply:
I meant: `get :index`